File: class2relational/src/main/java/org/atlanmod/scratch/classModel/ClassMetamodel.java
package org.atlanmod.scratch.classModel;

import org.atlanmod.tl.model.Metamodel;
import scala.Option;

public class ClassMetamodel implements Metamodel<ClassElement, ClassLink, String, String> {

    @Override
    public Option<ClassElement> toModelClass(String sc, ClassElement se) {
        // TODO
        return null;
    }

    @Override
    public Option<ClassLink> toModelReference(String sr, ClassLink sl) {
        // TODO
        return null;
    }
}

File: sparktl-transformations/src/main/scala/org/atlanmod/class2relational/model/classmodel/metamodel/ClassMetamodel.scala
package org.atlanmod.class2relational.model.classmodel.metamodel

import org.atlanmod.class2relational.model.classmodel.ClassModel
import org.atlanmod.class2relational.model.classmodel.element.{ClassAttribute, ClassClass, ClassClassifier}
import org.atlanmod.tl.model.impl.dynamic.{DynamicElement, DynamicLink, DynamicMetamodel}

trait ClassMetamodel extends Serializable {

    final val CLASS = "Class"
    final val ATTRIBUTE = "Attribute"
    final val DATATYPE = "Datatype"
    final val CLASS_ATTRIBUTES = "attributes"
    final val ATTRIBUTE_CLASS = "owner"
    final val ATTRIBUTE_TYPE = "type"

    def metamodel: DynamicMetamodel[DynamicElement, DynamicLink] = new DynamicMetamodel[DynamicElement, DynamicLink]("ClassMetamodel")

    def getAttributeType(attribute: ClassAttribute, model: ClassModel): Option[ClassClassifier]

    def getAttributeOwner(attribute: ClassAttribute, model: ClassModel): Option[ClassClass]

    def getClassAttributes(cl: ClassClass, model: ClassModel): Option[List[ClassAttribute]]
}
